pkg/kubelet/kubelet_node_status.go

```diff
@@ -739,12 +742,19 @@ func (kl *Kubelet) defaultNodeStatusFuncs() []func(context.Context, *v1.Node) er
 	// Volume limits
 	setters = append(setters, nodestatus.VolumeLimits(kl.volumePluginMgr.ListVolumePluginWithLimits))
 
+	var readyFuncs = []func() error{
+		kl.runtimeState.runtimeErrors,
+		kl.runtimeState.networkErrors,
+		kl.runtimeState.storageErrors,
+		kl.shutdownManager.ShutdownStatus,
+		validateWritableDirectory("root", kl.getRootDir()),
```

Comment:
I went back and forth on which directories to check (e.g. just root, or just pods, or every writable directory).  I'm sort of leaning towards just checking the pods directory as it would catch most problems and putting portions of the kubelet root on different filesystems is probably rare.